src/main/scala/at/logic/skeptik/algorithm/prover/package.scala

```diff
@@ -0,0 +1,192 @@
+package at.logic.skeptik.algorithm
+
+import at.logic.skeptik.algorithm.prover.structure.immutable.Literal
+import at.logic.skeptik.algorithm.unifier.{MartelliMontanari => unify}
+import at.logic.skeptik.expression.substitution.immutable.Substitution
+import at.logic.skeptik.expression.{Abs, App, E, Var, i}
+import at.logic.skeptik.judgment.immutable.SeqSequent
+
+import scala.collection.mutable
+import scala.language.implicitConversions
+
+/**
+  * @author Daniyar Itegulov
+  */
+package object prover {
+  type Clause = SeqSequent
+  type CNF = structure.immutable.CNF
+
+  object Clause {
+    def apply(a: E*)(b: E*) = new SeqSequent(a, b)
+    def empty = SeqSequent()()
+  }
+
+  implicit def varToLit(variable: E): Literal = Literal(variable, negated = false)
+
+  implicit def literalToClause(literal: Literal): Clause = literal.toClause
+
+  implicit class ClauseOperations(clause: SeqSequent) {
+    lazy val literals: Seq[Literal] =
+      clause.ant.map(Literal(_, negated = true)) ++ clause.suc.map(Literal(_, negated = false))
+
+    def apply(pos: Int): Literal = literals(pos)
+
+    def first: Literal = apply(0)
+
+    def last: Literal = apply(literals.length - 1)
+
+    def isUnit: Boolean = clause.width == 1
+  }
+
+  implicit class UnitSequent(sequent: SeqSequent) {
```

Comment:
To improve performance, it is a good idea to make implicit classes extend "AnyVal".  Google about it to learn more.